src/components/permissions/hooks/usePermissions.js

```diff
@@ -41,6 +41,16 @@ const usePermissions = (permission: Permission, options = {}) => {
   const PromptPopup = promptPopup || promptPopups[permission]
   const DeniedPopup = deniedPopup || deniedPopups[permission]
 
+  const initializeAllowed = useCallback(async () => {
```

Comment:
remove this. allowed check is performing in the handleRequestFlow. first check then prompt and request (if need)

Reply:
done